### ISSUE TITLE
Error in vis_hyperparams.py

### ISSUE THREAD

**Issue by liuweie** [zyhbili/Pose2Img#4]
hello,when I use `vis_hyperparams.py` to compute `kp_var_root` and `mask_sigma_perp`  ,the following error ocurred:

![image](https://user-images.githubusercontent.com/86940610/204984448-1cac9644-12cc-4ee9-b372-8c82d3186ee5.png)

I've been debugging for a long time and still can't solve it. Can you help me to see what the problem is? .

**Comment by zyhbili** [comment]
Have you tried the demo data and cfg?  It seems the problem of your custom keypoints, are they of shape (3,137)?

**Comment by liuweie** [comment]
In fact，the image showed above is exactly demo data of oliver,and oliver's keypoints shape is (122, 2):
`import numpy as np`
`path = 'D:\\Pose2Img\\data\\Oliver\\keypoints\\Auto_Lending_-_Last_Week_Tonight_with_John_Oliver_HBO-4U2eDJnwz_s_58_000_0_00_05.405405.npy'`
`npy = np.load(path)`
`print(npy.shape)`

and the result:
`(122, 2)`

**Comment by zyhbili** [comment]
please pull the code

**Comment by liuweie** [comment]
Oh！I tried keypoints of shape(3,137), problem solved now, thanks for your patience !

**Comment by liuweie** [comment]
May I ask can you provide your email address hhh? Now I have basically implemented your project according to your code, but the current effect is not very good. I want to show you the demo I have made, and I hope you can give me some suggestions. (It's not convenient to post personal photos on GitHub because they involve individuals), It doesn't matter if it's inconvenient for you，thanks again.

**Comment by liuweie** [comment]
May I ask what are the specific meanings of the two parameters `kp_var_root` and `mask_sigma_perp`? Is there any explanation in the paper?And here are some of my img result using vis_hyperparams.py ,does this prove that this is a good parameter setting?
![5](https://user-images.githubusercontent.com/86940610/205799018-96411d1e-5d0a-4f4d-b8fa-c60c414535fb.png)

**Comment by liuweie** [comment]
![1](https://user-images.githubusercontent.com/86940610/205799896-6294b745-f0a7-4cb0-8e92-fa7a7a62a1ba.png)

**Comment by liuweie** [comment]
![2](https://user-images.githubusercontent.com/86940610/205800057-451d26c6-3664-4582-8558-f88a6b2939d5.png)

**Comment by zyhbili** [comment]
The two parameters are used to control the size of gaussian map. The keypoint gaussian maps are used to indicate the target pose. The per-part mask gaussian maps are used to fetch the source features and they will be refined in the sub module. You can refer to [Synthesizing Images of Humans in Unseen Poses](https://openaccess.thecvf.com/content_cvpr_2018/CameraReady/1978.pdf) for more details. 
It seems a proper hyperparams setting. If you produce poor results, you can augment your custom training data with more diverse poses to bridge the gap between train and test distribution. For further discussing, you could contact me by email zhiyh@shanghaitech.edu.cn.

**Comment by liuweie** [comment]
Thanks very much!
